lib/common_test/src/ct_suite.erl

```diff
@@ -0,0 +1,132 @@
+-module(ct_suite).
+
+%%------------------------------------------------------------------
+%% Test Suite Behaviour
+%% ------------------------------------------------------------------
+-export_type([ct_testname/0,
+              ct_groupname/0,
+              ct_config/0,
+              ct_status/0,
+              ct_groups_def/0,
+              ct_group_props/0,
+              ct_testcase_ref/0,
+              ct_group_ref/0,
+              ct_subgroups_def/0,
+              ct_tests_def/0,
+              ct_info/0
+             ]).
+
+-type ct_testname() :: atom().
+-type ct_groupname() :: atom().
+-type ct_config() :: [{Key :: atom(), Value :: term()}].
+-type ct_status() :: ok |
+            skipped |
+            failed.
+-type ct_group_props() :: [
+                parallel |
+                sequence |
+                shuffle |
+                {shuffle, Seed :: {integer(), integer(), integer()}} |
+                {ct_group_repeat_type(), ct_test_repeat()}
+            ] |
+            default.
```

Comment:
This option (`default`) isn't documented as valid for [`common_test Module:groups/0`](https://erlang.org/doc/man/common_test.html#Module:groups-0). If it comes to it I'll have to segregate it - along with a bunch of other types.

Reply:
It is not valid for `Module:groups`, so the `group_props` for `group_ref` is different from the `group_props` of `group_def`. Which means that they have to be segregated. I would suggest creating a type called `group_props_ref` that is `group_props() | default`, or something along those lines.

Reply:
I'll "fix" it but it'll involve more than just (simply) a new type.

Reply:
How does 40d021e look?